x-pack/plugins/graph/public/helpers/use_workspace_loader.ts

```diff
@@ -74,14 +74,41 @@ export const useWorkspaceLoader = ({
       store.dispatch({ type: 'x-pack/graph/RESET' });
     }
 
+    async function* pageThroughIndexPatterns() {
+      let perPage = 1000;
+      let total = 0;
+      let savedObjects: IndexPatternSavedObject[] = [];
+
+      async function* makeRequest(page: number): AsyncGenerator<IndexPatternSavedObject[]> {
+        await savedObjectsClient
+          .find<{ title: string }>({
+            type: 'index-pattern',
+            fields: ['title', 'type'],
+            perPage,
+            page,
+          })
+          .then((response) => {
+            perPage = response.perPage;
+            total = response.total;
```

Comment:
Nit: shouldn't be necessary to set `perPage` since it shouldn't change, which means it could be a `const`